src/foam/nanos/auth/ProfilePictureView.js

```diff
@@ -132,14 +192,62 @@ foam.CLASS({
     function onAddAttachmentClicked (e) {
       this.document.querySelector('.attachment-input').click();
     },
-
+    function onRemoveClicked (e) {
+      this.isDrag = false;
+      this.data = null;
+    },
+    function onDragOver(e) {
+      this.isDrag = true;
+      e.preventDefault();    
+    },
+    function onDrop(e) {
+      e.preventDefault();  
+      this.isDrag = false;
+      if(this.uploadHidden){
+        return;
+      }
+      else{
+        var inputFile;
+        if (e.dataTransfer.items) {
+          inputFile = e.dataTransfer.items[0]
+          if (inputFile.kind === 'file') {     
+            var file = inputFile.getAsFile();
+            if(this.isRightType(file)){
+              this.addFile(file);
+            }
+            else{
+              this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' }));
+            } 
+          }
+        }else if(e.dataTransfer.files){
+          var file = e.dataTransfer.files[0];
+          if(this.isRightType(file)) 
+            this.addFile(file);
+          else{
+            this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' }));
+          }  
+        }
+      }
+    },
+    function isRightType(file){
```

Comment:
Perhaps isImageType would be a better name.

Better yet do we really care what type of file it is?  Maybe just checking that it starts with image/ is sufficient.